packages/@uppy/companion/src/server/helpers/request.js

```diff
@@ -79,15 +79,23 @@ function isPrivateIP (ipAddress) {
 
 module.exports.FORBIDDEN_IP_ADDRESS = FORBIDDEN_IP_ADDRESS
 
-module.exports.getRedirectEvaluator = (requestURL, blockPrivateIPs) => {
-  const { protocol } = new URL(requestURL)
+module.exports.getRedirectEvaluator = (rawRequestURL, blockPrivateIPs) => {
+  const requestURL = new URL(rawRequestURL)
   return (res) => {
     if (!blockPrivateIPs) {
       return true
     }
 
     const redirectURL = res.headers.location
-    const shouldRedirect = redirectURL ? new URL(redirectURL).protocol === protocol : false
+    let shouldRedirect = false
+    if (redirectURL) {
+      try {
+        shouldRedirect = new URL(redirectURL, requestURL).protocol === requestURL.protocol
```

Comment:
Great to block redirects from https to http, but what about the safe way, from http to https?

Reply:
Also, what about minimizing the scope of the try/catch, so you only try the `new URL` constructor and the rest lives outside of it? That way you don't accidentally ignore other errors in this code block

Reply:
> Great to block redirects from https to http, but what about the safe way, from http to https?

that would probably be good, but i'd rather only fix the crash here so we can publish it quickly and not risk introducing new mistakes :D